WordPress/Classes/ViewRelated/Site Creation/Shared/TitleSubtitleTextfieldHeader.swift

```diff
@@ -87,16 +87,40 @@ final class SearchTextField: UITextField {
         addBottomBorder(withColor: .divider)
     }
 
-    private func setIconImage() {
+    private lazy var searchIconImageView: UIImageView = {
         let iconSize = CGSize(width: Constants.iconDimension, height: Constants.iconDimension)
         let loupeIcon = Gridicon.iconOfType(.search, withSize: iconSize).imageWithTintColor(.listIcon)?.imageFlippedForRightToLeftLayoutDirection()
-        let imageView = UIImageView(image: loupeIcon)
+        return UIImageView(image: loupeIcon)
+    }()
+
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        let activityIndicator: UIActivityIndicatorView
+        if #available(iOS 13, *) {
+            activityIndicator = UIActivityIndicatorView(style: .medium)
+        } else {
+            activityIndicator = UIActivityIndicatorView(style: .gray)
+        }
+        activityIndicator.backgroundColor = UIColor.clear
+
+        return activityIndicator
+    }()
+
+    func setIcon(isLoading: Bool) {
+        if isLoading {
+            setIconImage(view: activityIndicator)
+            activityIndicator.startAnimating()
+        } else {
+            activityIndicator.stopAnimating()
+            setIconImage(view: searchIconImageView)
+        }
+    }
 
+    private func setIconImage(view: UIView) {
         if traitCollection.layoutDirection == .rightToLeft {
```

Comment:
I've checked the outcome simulating a right-to-left language and the result was this:

<img src="https://user-images.githubusercontent.com/7040243/76235491-f00f8c80-6209-11ea-9a4a-027eaa0b47dd.png" width="300" />

The position of the search icon and the loading indicator is the same as left-to-right languages. I don't think this is a problem but I didn't get what this code is doing.

Or did I missed something?

Besides that, it works like a charm. :)